<launch>
  <node pkg="mlcc" type="global" name="global" output="screen">
    <!-- test scene -->
    <param name="data_path" type="string" value="$(find mlcc)/scene2/"/>

    <!-- optimization iteration number -->
    <param name="max_iter" type="int" value="10"/>

    <!-- base LiDAR: 0-2 is the MID from left to right, 3 is AVIA -->
    <param name="base_lidar" type="int" value="3"/>

    <!-- LiDAR to be calibrated
    The extrinsic sequence in ref.json needs to be the same with here -->
    <param name="ref_lidar1" type="int" value="0"/>
    <param name="ref_lidar2" type="int" value="1"/>
    <param name="ref_lidar3" type="int" value="2"/>

    <!-- largest voxel size:
    if AVIA is set as base LiDAR use 4, if MID is set as base LiDAR use 1 -->
    <param name="voxel_size" type="double" value="4"/>

    <!-- base LiDAR downsampling size:
    if AVIA is set as base LiDAR use 0.4, if MID is set as base LiDAR use 0.1 -->
    <param name="downsample_base" type="double" value="0.4"/>

    <!-- LiDAR to be calibrated downsampling size:
    if AVIA is set as base LiDAR use 0.2 (e.g., using AVIA to calibrate MID),
    if MID is set as base LiDAR use 0.1 (e.g., MID self calibration) -->
    <param name="downsample_ref" type="double" value="0.2"/>
    <param name="eigen_threshold" type="double" value="20"/>
  </node>

  <arg name="rviz" default="true" />
  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz"
    args="-d $(find mlcc)/rviz_cfg/rosbag.rviz" />
  </group>
</launch>